woocommerce/woocommerce-android
# Free Trial status bar

Closes: #8375


### Description

This PR adds a banner informing users that they're in the trial.

### Testing instructions

TC 1: Show/hide between sites
1. Select a Free Trial site
2. Assert that banner is visible
3. Change to a non-Free Trial site
4. Assert that banner is gone
5. Repeat the test cast

TC 2: Upgrade flow
1. Select a Free Trial site
2. Tap on "Upgrade now" button on the banner
3. Assert that you are not requested to provide any credentials, and you can see the upgrade flow site.
4. Upgrade the site a the Commerce plan (the easiest: with credits)
5. Asser that after upgrade, you're redirected to the previous app's screen and trial banner is gone

### Images/gif

https://user-images.githubusercontent.com/5845095/224015405-35eea233-a9c8-4354-a55d-f59ae6bbad1b.mp4
